### PR TITLE
Refector of shell to act as intepreter

### LINKED ISSUE #10: shell key in schema
@vsoch 

Inspired by GitHub action https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell i think the `shell` key would be useful if we change its behavior. Let's face it, our current implementation of having `shell` can be achieved with using `run` key so I'd propose we change the behavior of `shell` key to mimic the GitHub action. Not all the features but some of it. The default shell can be `bash` but it can be changed
 
The script below will run in `bash` shell it is assumed `shell: bash`  and key is not required.

```
version: 0.0.1
shell_check:
  type: script
  run: |
    echo $SHELL
```

This allows user to write test in any shell so for instance they can write a test for `csh` as follows

```
version: 0.0.1
shell_check:
  type: script
  shell: csh
  run: |
    echo $SHELL
    setenv PATH $PATH:\/home/user/bin
    printenv  
```

Next we can extend the `run` key to accept python code if `shell: python` so it makes it easy to write scripts. Setting `shell: python` will use the python interpreter.

```
version: 0.0.1
shell_check:
  type: script
  shell: python
  run: |
    import os
    print os.cwd    
```

We can decide for python interpreter to either default to the following

1.  /usr/bin/python
2. python wrapper in $PATH

If we decide with **1.** then we can avoid unexpected results with user environment (virtualenv, conda, pip, module) that can affect python wrapper when running buildtest. If we go with **1.** then we are writing in Python 2 assuming ``/usr/bin/python`` exist in the system. If it doesn't it should through an error which is fine, because we expect same behavior if a shell is not available lets say ``/bin/csh`` is not available then ``shell: csh`` should error out.

If we go with **1.** then we may need ``shell: python3`` to accept python 3 script in this case ``/usr/bin/python3``. 

If we go with **2.** then we don't have to worry about system python but it can cause confusion to user if it is accepting python 2 or python 3. 

For now i think it is sufficient to allow the follow values

``shell: bash``, ``shell: csh``, ``shell: sh``, ``shell: python``, ``shell: python3``

### PR DESCRIPTION
**Pull request by vsoch**
This will refactor the shell parameter to act more as an interpreter instead of a command line. Run should be used for commands, with shell designating the interpreter to use. This will close #10.

### CODE REVIEW

**Comment by vsoch** [comment]
@shahzebsiddiqui should the default shell interpreter be sh instead of bash? This is the case for Singularity. On the one hand, it will support old legacy systems better. On the other hand, most people use bash these days, and it means they would need to set it every time. I chose bash for this reason - I think the majority of people would want it over the (non bourne again) shell.

File: README.md [hunk 1]
```diff
@@ -102,6 +102,37 @@ Adding a new version means that you only need to:
...
+## Development Tips
+
+### Multiple Types
+
+See how [multiple types](https://cswr.github.io/JsonSchema/spec/multiple_types/) are handled.
+Basically, this snippet says that for shell the user can provide a string or an array (a list
+of strings or commands to execute). The variables `items`, and `minItems` apply only
+in the case that an array is used, and `minLength` applies to commands.
+
+```json
```
Comment (shahzebsiddiqui) [inline README.md]: I think this doc example is misleading. `shell` is of type string not array with the recent change.

File: script/examples/0.0.1/slurm.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version: 0.0.1
 # Shell would be accepted to indicate a single line shell command (or similar)
 login_node_check:
   type: script
-  shell: "ping login 1"
+  run: "ping login 1"
```
Comment (shahzebsiddiqui) [inline script/examples/0.0.1/slurm.yml]: Not sure about the YAML syntax i think it should be ok to have ``run: "ping login 1" or should we be consistent
and have
```
  run: |
ping login  1
```

**Comment by shahzebsiddiqui** [comment]
I think the use case for Singularity is different. I would prefer bash as
most users are used to it. For most end users they probably don't know or
care difference between sh and bash. But certainly folks that use cash/tcsh
will need a way to run their scripts.

For now we can add support for bash. Then we can follow up with other
shells.

File: README.md (hunk 1 above)
Comment (vsoch) [reply README.md]: The README in the top level is only to provide examples to help developers, that's not the current implementation, but I undoubtedly will need to remember how to do that, which is why it's under a section called "Development Tips" and then "Multiple Types." The docs about the script command are in the README.md of the script folder, which is not this file.

File: script/examples/0.0.1/slurm.yml (hunk 2 above)
Comment (vsoch) [reply script/examples/0.0.1/slurm.yml]: Both of those cases are strings, which is what run accepts. The only difference is that the second (with the pipe) is just yaml synax for allowing more than one line.

File: README.md (hunk 1 above)
Comment (shahzebsiddiqui) [reply README.md]: Make sense to me now.

**Comment by vsoch** [comment]
Thanks!

**Pull request merged**